staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/controller.go

```diff
@@ -231,31 +246,40 @@ func (c *celAdmissionController) Validate(
 			// be a bindingInfo for it
 			bindingInfo := c.bindingInfos[namespacedBindingName]
 			binding := bindingInfo.lastReconciledValue
-			if !binding.Matches(a) {
+			matches, err := c.validatorCompiler.BindingMatches(a, o, binding)
+			if err != nil {
+				return err
+			}
+			if !matches {
 				continue
 			}
 
 			var param *unstructured.Unstructured
 
-			// If definition has no paramsource, always provide nil params to
+			// If definition has no paramKind, always provide nil params to
 			// evaluator. If binding specifies a params to use they are ignored.
 			// Done this way so you can configure params before definition is ready.
```

Comment:
I guess the comments here are misleading. We current ask if paramKind set in definition, paramRef has to be set; if paramKind is not set in definition, paramRef should not be set. Behavior against this will be treated as mis configuration and lead to an err. Please let me know if other behavior is expected

Reply:
Tracked in below comment